### PR TITLE
SQL: [Docs] Add example for custom bucketing with CASE

### PR DESCRIPTION
**Pull request by matriv**
Add a TIP on how to use CASE to achieve custom bucketing
with GROUP BY.

Follows: #41349

### CODE REVIEW

File: docs/reference/sql/functions/conditional.asciidoc [hunk 1]
```diff
@@ -98,6 +98,31 @@ an error message would be returned, mentioning that *'foo'* is of data type *key
 which does not match the expected data type *integer* (based on result *10*).
 ===============================
 
+[[sql-functions-conditional-case-groupby-custom-buckets]]
+[TIP]
+===============================
+CASE can be used as a GROUP BY key in a query to facilitate custom bucketing
+and assign descriptive names to those buckets. If for example the number of
+values for a key are too many or, simply, ranges of those values are more
```
Comment (astefan) [inline docs/reference/sql/functions/conditional.asciidoc]: `the number .....` is too large. `values.... are too many`

File: docs/reference/sql/functions/conditional.asciidoc [hunk 2]
```diff
@@ -98,6 +98,31 @@ an error message would be returned, mentioning that *'foo'* is of data type *key
 which does not match the expected data type *integer* (based on result *10*).
 ===============================
 
+[[sql-functions-conditional-case-groupby-custom-buckets]]
+[TIP]
+===============================
+CASE can be used as a GROUP BY key in a query to facilitate custom bucketing
+and assign descriptive names to those buckets. If for example the number of
```
Comment (astefan) [inline docs/reference/sql/functions/conditional.asciidoc]: `If, for example, the number....`

File: docs/reference/sql/functions/conditional.asciidoc [hunk 3]
```diff
@@ -98,6 +98,31 @@ an error message would be returned, mentioning that *'foo'* is of data type *key
...
+    WHEN languages = 1 THEN 'one'
+    WHEN languages = 2 THEN 'bilingual'
+    WHEN languages = 3 THEN 'trilingual'
+    ELSE 'multilingual'
+  END as lang_skills
+FROM employees
+GROUP BY lang_skills
+ORDER BY lang_skills;
+
+With this query, we can create normal grouping buckets for values _0, 1, 2, 3_ with
```
Comment (astefan) [inline docs/reference/sql/functions/conditional.asciidoc]: `With this query, one can create`

**Review by astefan — APPROVED** [review]
LGTM. Left few minor cosmetic comments.

File: docs/reference/sql/functions/conditional.asciidoc [hunk 4]
```diff
@@ -98,6 +98,31 @@ an error message would be returned, mentioning that *'foo'* is of data type *key
 which does not match the expected data type *integer* (based on result *10*).
 ===============================
 
+[[sql-functions-conditional-case-groupby-custom-buckets]]
+[TIP]
```
Comment (costin) [inline docs/reference/sql/functions/conditional.asciidoc]: I would remove the TIP header and promote this to a section (`=====`) - I think it's big enough to warrant a proper entry.

**Review by costin — APPROVED** [review]
Left a comment - this is cool enough that it deserves its own section.

**Pull request merged**